src/bin/pgcopydb/schema.c

```diff
@@ -2628,22 +2628,27 @@ schema_list_partitions(PGSQL *pgsql, SourceTable *table, uint64_t partSize)
 
 	char *sqlTemplate =
 		" with "
-		" t(min, max, parts) as "
+		" key_bounds (min, max) as "
 		" ( "
-		"   select min(\"%s\"), max(\"%s\"), "
-		"          ceil(pg_table_size('%s.%s')::float / $1) "
-		"     from \"%s\".\"%s\""
-		"  ), "
-		"  ranges(n, parts, a, b) as "
-		"  ( "
-		"    select n, "
-		"           parts, "
-		"           x as a, "
-		"           coalesce((lead(x, 1) over(order by n)) - 1, max) as b "
-		"      from t, "
-		"           generate_series(min, max, ((max-min+1)/parts)::bigint + 1) "
-		"           with ordinality as s(x, n) "
-		"  ) "
+		"   select min(\"%s\"), max(\"%s\") "
+		"     from \"%s\".\"%s\" "
+		" ), "
+		" t (parts) as "
+		" ( "
+		"   select greatest(ceil(bytes::float / $1), 1) "
+		"     from pgcopydb.table_size "
+		"     where oid = %d "
+		" ), "
```

Comment:
I have reduced one space ident here. Let me know if this needs to be reverted.

Reply:
Does the subquery returns a value when the WHERE clause fails to match any row in the pgcopydb.table_size table? Maybe this warrants an additional test-case in the `tests/unit` test suite...

Reply:
I have revised this, please take a look.

Reply:
It's better to use UNION ALL in that case, because it skips the deduplication step that Postgres would have to implement with using UNION: we know we are not introducing duplicate data here. Of course in terms of performance, it is not going to be dramatic given we expect only one row in both branches of the UNION. It's more to do with semantics here.